tiled/adapters/arrow.py

```diff
@@ -0,0 +1,322 @@
+        """
+        Function to initialize and return the reader handle.
+        Returns
+        -------
+        The reader handle.
+        """
+        partitions = []
+        for path in self._partition_paths:
+            if not Path(path).exists():
+                partition = None
+            else:
+                with pyarrow.ipc.open_file(path) as reader:
+                    partition = reader
+            partitions.append(partition)
+        return partitions
+
+    def write_partition(
+        self,
+        data: Union[List[pyarrow.record_batch], pyarrow.record_batch, pandas.DataFrame],
+        partition: int,
+    ) -> None:
+        """
+        "Function to write the data into specific partition as arrow format."
+        Parameters
+        ----------
+        data : data to write into arrow file. Can be a list of record batch, or pandas dataframe.
+        partition: integer index of partition to be read.
+        Returns
+        -------
+        """
+        if isinstance(data, pandas.DataFrame):
+            table = pyarrow.Table.from_pandas(data)
+            batches = table.to_batches()
+        else:
+            if not isinstance(data, list):
+                batches = [data]
+            else:
+                batches = data
+
+        schema = batches[0].schema
+
+        uri = self._partition_paths[partition]
+
+        with pyarrow.ipc.new_file(uri, schema) as file_writer:
+            for batch in batches:
+                file_writer.write_batch(batch)
+            file_writer.close()
+
+    def write(
+        self,
+        data: Union[List[pyarrow.record_batch], pyarrow.record_batch, pandas.DataFrame],
+    ) -> None:
+        """
+        "Function to write the data as arrow format."
+        Parameters
+        ----------
+        data : data to write into arrow file. Can be a list of record batch, or pandas dataframe.
+        Returns
+        -------
+        """
+        if isinstance(data, pandas.DataFrame):
+            table = pyarrow.Table.from_pandas(data)
+            batches = table.to_batches()
+        else:
+            if not isinstance(data, list):
+                batches = [data]
+            else:
+                batches = data
+
+        schema = batches[0].schema
+
+        if self.structure().npartitions != 1:
+            raise NotImplementedError
+        uri = self._partition_paths[0]
+
+        with pyarrow.ipc.new_file(uri, schema) as file_writer:
+            for batch in data:
+                file_writer.write_batch(batch)
+            file_writer.close()
+
+    def read(self, *args: Any, **kwargs: Any) -> pandas.DataFrame:
+        """
+        The concatenated data from given set of partitions as pyarrow table.
+        Parameters
+        ----------
+        Returns
+        -------
+        Returns the concatenated pyarrow table as pandas dataframe.
+        """
+        if any(p is None for p in self.reader_handle):
```

Comment:
This is inefficient because it tries to open all the files and _then_ checks whether any are missing. In fact, as soon as we find one missing we can stop the loop and raise an error.